gym_roboy/envs/roboy_env.py

```diff
@@ -57,24 +51,19 @@ def _create_reward_range(self, robot: RoboyRobot) -> Tuple[float, float]:
     def step(self, action):
         assert self.action_space.contains(action)
 
-        if self._is_tendon_vel_dependent_on_distance:
-            distance_to_target = _l2_distance(self._last_state.joint_angles, self._goal_state.joint_angles)
-            scale_factor = np.power(distance_to_target / self._MAX_DISTANCE_JOINT_ANGLE, 1/4)
-            current_max_tendon_speed = scale_factor * self._robot.get_action_space().high
-            action = np.multiply(current_max_tendon_speed, action)
-
-        else:
-            action = np.multiply(self._robot.get_action_space().high, action)
-
-        action = action.tolist()
+        action = _rescale_from_one_space_to_other(
+            input_space=self.action_space,
+            output_space=self._robot.get_action_space(),
+            input_val=np.array(action)).tolist()
 
         new_state = self._ros_proxy.forward_step_command(action)
         self.step_num += 1
         self._last_state = new_state
         obs = self._make_obs(robot_state=new_state)
         info = {}
         reward = self.compute_reward(current_state=new_state, goal_state=self._goal_state, info=info)
-        done = self._did_complete_successfully(current_state=new_state, goal_state=self._goal_state) or self.step_num > self._MAX_EPISODE_LENGTH
+        done = self._did_complete_successfully(current_state=new_state, goal_state=self._goal_state) or \
+               self.step_num > self._MAX_EPISODE_LENGTH
```

Comment:
let us set the `done` variable with a positive / success connotation.
Therefore, we should divide the condition when `done` can be `true`.

I am thinking about introducing a new variable like `failure`, which is set to `true`, when the agent has not reached the goal state within a certain amount of steps.

Reply:
`done` as a return value of the `gym.Env.step(...)` method has this description in the docs: `"done: whether the episode has ended, in which case further step() calls will return undefined results".` That's why `done` can also be true, even if our agent has failed to reach the target, but surpassed the maximum episode length.

I can definitely factor the second part of the or condition, and since `failure` is completely determined by the variable `self.step_num`, maybe we can implement a method `has_failed()` rather than an attribute `failure`.

For now, the only reason for failure is if the agent surpasses the maximum episode length. What about the name `ran_out_of_time()`, which is more specific?

Reply:
Thanks for your comments. They were helpful.

What about `reached_max_steps()` ?
Time is relative.

Reply:
@ALPA-Industry-and-Technology great name